### ISSUE TITLE
Report fields - Initial value in the system message is defined as "undefined"

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#44885]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.4-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:**  Applause-Internal team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to workspace settings > Report fields.
3. Create a List type report field, add a few list values and select an initial value (important).
4. Go to workspace chat.
5. Submit two expenses.
6. Go to expense report.
7. Click on the List report field created in Step 3.
8. Select other item other than the initial value.
9. Revisit the expense report.

## Expected Result:
The system message for the initial value should be defined properly.

## Actual Result:
The system message for the initial value is "undefined" - "change field from undefined to X".

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/669fde27-36a3-44f9-8ef3-11685cde9509

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mountiny** [comment]
I can take this over and look into it with the project team

@waterim @rezkiy37 @shubham1206agra @rushatgabhane

**Comment by mountiny** [comment]
Not a blocker because this is behind beta

**Comment by izarutskaya** [comment]
We think this issue might be related to the  #vip-vsb

**Comment by nkdengineer** [comment]
@mountiny Is this external for contributor or the project team will take care this issue?

**Comment by mountiny** [comment]
@nkdengineer Lets do external for this one

**Comment by nkdengineer** [comment]
This is a BE bug, when we change the report field the first time `originalMessage` doesn't have `oldValue`.


<img width="573" alt="Screenshot 2024-07-05 at 18 00 05" src="https://github.com/Expensify/App/assets/161821005/a4c819f6-73ed-4f17-be37-096ba764b033">

**Comment by mountiny** [comment]
@nkdengineer I think there might be a different issue then. When the expense is created, they should have the initial value for the report field set, right? Can you confirm that this is happening? 

Also I can see that we have this logic https://github.com/Expensify/App/blob/16c2f5350b1f30684ec2768aa403fdbabf84bce8/src/libs/ReportActionsUtils.ts#L1240-L1243

Can we update this that if the oldValue is undefined, we use the `set` copy

**Comment by nkdengineer** [comment]
@mountiny You're right, the problem here is we missed to early return here so we still move to the below case which display the translation with `oldValue` as `undefined`.

We should early return here and the bug can be fixed.

https://github.com/Expensify/App/blob/16c2f5350b1f30684ec2768aa403fdbabf84bce8/src/libs/ReportActionsUtils.ts#L1240-L1243

**Comment by nkdengineer** [comment]
@mountiny I can raise the PR to fix now if I assigned.

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The system message for the initial value is "undefined" - "change field from undefined to X".

### What is the root cause of that problem?
1. We have a BE bug here https://github.com/Expensify/App/issues/44885#issuecomment-2210674158

2. We missed to early return when `oldValue` is empty and then `undefined` is displayed


https://github.com/Expensify/App/blob/c80e8b65eb8bb5f218d50a59efb2a65f22d7c527/src/libs/ReportActionsUtils.ts#L1173-L1175
### What changes do you think we should make in order to solve the problem?
1. For BE bug we can fix as a separate issue

2. We should early return here if `oldValue` is empty

```
if (!oldValue) { 
   Localize.translateLocal('report.actions.type.changeFieldEmpty', {newValue, fieldName});
   return;
} 
```

https://github.com/Expensify/App/blob/c80e8b65eb8bb5f218d50a59efb2a65f22d7c527/src/libs/ReportActionsUtils.ts#L1173-L1175
### What alternative solutions did you explore? (Optional)

**Comment by mountiny** [comment]
@nkdengineer that is good, however, can you explore how this compares with creating the expense on OldDot with the same settings? is the report field on the expense populated with whatever the initial value was? if yes, we need to make sure expense created in NewDot is the same.

**Comment by nkdengineer** [comment]
@mountiny The initial value of report field is correct. We only have a problem with the change field report action as I mentioned here https://github.com/Expensify/App/issues/44885#issuecomment-2210674158 when we edit the report field at the first time.


<img width="575" alt="Screenshot 2024-07-06 at 01 03 55" src="https://github.com/Expensify/App/assets/161821005/05681f7c-0536-4e40-b29c-f6d84af45651">

**Comment by mountiny** [comment]
Correct, although I checked the BE and we are just pulling the existing value for the report field for the oldValue so if its not coming through and its undefined, it must not have been set 🤔

**Comment by mountiny** [comment]
@nkdengineer lets go with your proposal to make sure we treat the oldValue as empty when undefined

**Comment by nkdengineer** [comment]
@mountiny Do I need to create a PR here?

**Comment by mountiny** [comment]
@nkdengineer I have confirmed what I wanted to check. I'm not sure what was going on, but yesterday, all your posts disappeared; I went ahead and created the quick PR. We would pay you $50 for your help

**Comment by nkdengineer** [comment]
Thanks. I don't know what the problem is, but on my side everything is good and it's too late at the time you assigned to me.

**Comment by mountiny** [comment]
I know, but your gh profile was showing 404 not found and all your comments diseappeared :D github was doing april fools, I guess

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mountiny** [comment]
This is ready for payment, $50 to @nkdengineer

**Comment by sakluger** [comment]
@nkdengineer I sent you an offer via Upwork: https://www.upwork.com/nx/wm/offer/103223949/

Please let me know once you've accepted it. Thanks!

**Comment by nkdengineer** [comment]
@sakluger I accepted it thanks
